eeprom.c

```diff
@@ -157,25 +157,101 @@ int start_programming(llio_t *self, u32 start_address, int fsize) {
     return 0;
 }
 
-int eeprom_write(llio_t *self, char *bitfile_name, u32 start_address, int fix_boot_flag) {
+int eeprom_write(llio_t *self, char *bitfile_name, u32 start_address, int fix_boot_flag, int sha256_check_flag) {
     board_t *board = self->board;
-    int bytesread, i;
+    int bytesread, i, j;
     u32 eeprom_addr;
     char part_name[32];
     struct stat file_stat;
     FILE *fp;
     struct timeval tv1, tv2;
+    char sha256str[65];
+    char sha256file_name[262];
+    unsigned char sha256in[32];
+    unsigned char sha256bitfile[32];
+    SHA256_CTX sha256ctx;
 
-    if (stat(bitfile_name, &file_stat) != 0) {
-        printf("Can't find file %s\n", bitfile_name);
-        return -1;
-    }
+    if (sha256_check_flag) {
+        printf("\nFPGA configuration bitfile integrity verification:\n");
+        strcpy(sha256file_name, bitfile_name);
+        strcat(sha256file_name, ".sha256");
```

Comment:
Please use truncating APIs, such as
```c
snprintf(sha256file_name, sizeof(sha256file_name), "%s.sha256", bitfile_name);
```
otherwise, if a filename passed in is long, the `sha256file_name` stack variable can be corrupted and lead to undefined behavior.